src/Simplex/Messaging/Agent/Stats.hs

```diff
@@ -0,0 +1,402 @@
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+module Simplex.Messaging.Agent.Stats where
+
+import qualified Data.Aeson.TH as J
+import Data.Map (Map)
+import Data.Time.Clock (UTCTime (..))
+import Simplex.Messaging.Agent.Protocol (UserId)
+import Simplex.Messaging.Parsers (defaultJSON)
+import Simplex.Messaging.Protocol (SMPServer, XFTPServer)
+import UnliftIO.STM
+
+data AgentSMPServerStats = AgentSMPServerStats
+  { fromTime :: TVar UTCTime,
+    msgSent :: TVar Int, -- total messages sent to server directly
+    msgSentRetries :: TVar Int, -- sending retries
+    msgSentSuccesses :: TVar Int, -- successful sends
+    msgSentAuth :: TVar Int, -- send AUTH errors
+    msgSentQuota :: TVar Int, -- send QUOTA errors
+    msgSentExpired :: TVar Int, -- send expired errors
+    msgSentErr :: TVar Int, -- other send errors (excluding above)
```

Comment:
not clear how they are related

Reply:
msgSent is total messages set for delivery
msgSent would be msgSentSuccesses + all errs + msgs that are in progress (e.g. may be retried, but msgSentRetries in general is greater than that - it's overall number of retries, not number of retried messages)